pkg/worker/gpu_info.go

```diff
@@ -0,0 +1,125 @@
+package worker
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+type GPUInfoClient interface {
+	AvailableGPUDevices() ([]int, error)
+	GetGPUMemoryUsage(deviceIndex int) (GpuMemoryUsageStats, error)
+}
+
+type GpuMemoryUsageStats struct {
```

Comment:
Nitpick, can we change Gpu to GPU for this?

Reply:
done